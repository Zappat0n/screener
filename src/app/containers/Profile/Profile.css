.profile {
  font-family: 'Open Sans', sans-serif;
}

.profile .content {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.profile .top {
  display: flex;
  align-items: center;
}

.profile .top .logo {
  width: 64px;
}

.profile .top h2, .profile .top h4 {
  margin-left: 2rem;
}
.profile .top h5 {
  margin-left: 0.25rem;
}

.profile .info {
  width: 75%;
}

.priceData {
  width: 60%;
}

.profile .row {
  display: flex;
  justify-content: space-between;
}

.profile .cell {
  width: 100%;
  margin: 0.5rem 1rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border-bottom: 1px solid lightgray;
}

.profile .info .cell {
  justify-content: space-evenly;
  border: none;
  font-weight: 700;
}
