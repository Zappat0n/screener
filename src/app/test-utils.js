import React from 'react';
import { createStore, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import indexReducer from './reducers/indexSlice';
import stockReducer from './reducers/stockSlice';
import profileReducer from './reducers/profileSlice';
import filterReducer from './reducers/filterSlice';

const rootReducer = combineReducers({
  indexes: indexReducer, stocks: stockReducer, profile: profileReducer, filter: filterReducer,
});

const initialState = {
  stocks: {
    AAPL: {
      price: 100,
      companyName: 'Apple Inc',
      changes: 1,
      sector: 'Technology',
    },
  },
  profile: {
    data: {
      image: '',
      companyName: 'Microsoft',
      price: 100,
      changes: 1,
      mktCap: 1000000,
      range: '0-100',
      country: 'US',
    },
  },
};

const renderComponent = (component) => render(
  <Provider store={createStore(rootReducer, initialState)}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export * from '@testing-library/react';
export { renderComponent };
