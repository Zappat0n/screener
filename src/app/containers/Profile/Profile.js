import { useDispatch, useSelector } from 'react-redux';
import { unwrapResult } from '@reduxjs/toolkit';
import { Link, useParams } from 'react-router-dom';
import { refreshProfile, updateProfile } from '../../reducers/profileSlice';
import './Profile.css';
import ProfileRow from '../../components/ProfileRow';

const Profile = () => {
  const { id } = useParams();
  const data = useSelector((state) => (state.profile.data ? state.profile.data : ''));
  const index = useSelector((state) => state.stocks.index);
  const dispatch = useDispatch();

  const initialize = () => {
    dispatch(refreshProfile(id))
      .then(unwrapResult)
      .then((originalPromiseResult) => {
        dispatch(updateProfile(originalPromiseResult));
      });
  };

  if (data.symbol !== id) {
    initialize();
  }

  const checkType = (data) => ((!data || typeof data === 'string') ? data : data.toString());

  return (
    <div className="profile">
      <div className="content">
        <Link key={index} to={`/index/${index || ''}`} className="link">
          Go Back
        </Link>
        <div className="top">
          <img className="logo" src={data.image} alt="logo" />
          <h2>
            {data.companyName}
          </h2>
          <h4>
            {data.price}
          </h4>
          <h5 className={data.changes < 0 ? 'colorRed' : 'colorBlue'}>
            {data.changes}
          </h5>
        </div>
        <div className="priceData">
          <ProfileRow title1="Previous close" data1={(Math.round((data.price - data.changes) * 100) / 100).toString()} title2="Market Cap" data2={checkType(data.mktCap)} />
          <ProfileRow title1="52 Week Range" data1={data.range} title2="Beta" data2={checkType(data.beta)} />
          <ProfileRow title1="Dividend Yield" data1={`${Math.round((data.lastDiv * 10000) / data.price) / 100}%`} title2="DCF" data2={checkType(data.dcf)} />
          <ProfileRow title1="Exchange" data1={data.exchangeShortName} title2="Country" data2={data.country} />
        </div>
        <div className="info">
          <h3>Company Information</h3>
          <ProfileRow title1="Sector:" data1={data.sector} title2="Industry:" data2={data.industry} />
          <h4>CEO</h4>
          <p>{data.ceo}</p>
          <h4>Website</h4>
          <p>{data.website}</p>
          <h4>Description</h4>
          <p>{data.description}</p>
        </div>
      </div>
    </div>
  );
};

export default Profile;
