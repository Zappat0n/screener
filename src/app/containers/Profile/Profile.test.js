import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import Profile from './Profile';
import { renderComponent, rendererForComponent } from '../../test-utils';

test('Renders Profile correctly', () => {
  renderComponent(
    <Profile ticker="MSFT" />,
  );
  expect(screen.getByText('Microsoft')).toBeInTheDocument();
  expect(screen.getAllByText('100')[0]).toBeInTheDocument();
  expect(screen.getByText('1')).toBeInTheDocument();
  expect(screen.getByText('1000000')).toBeInTheDocument();
  expect(screen.getByText('0-100')).toBeInTheDocument();
  expect(screen.getByText('US')).toBeInTheDocument();
});

test('Profile Match SnapShot', () => {
  const tree = rendererForComponent(
    <Profile ticker="MSFT" />,
  );
  expect(tree).toMatchSnapshot();
});
