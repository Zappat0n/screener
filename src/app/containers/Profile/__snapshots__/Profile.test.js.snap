// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Match SnapShot 1`] = `
<div
  className="profile"
>
  <div
    className="content"
  >
    <a
      className="link"
      href="/index/"
      onClick={[Function]}
    >
      Go Back
    </a>
    <div
      className="top"
    >
      <img
        alt="logo"
        className="logo"
        src=""
      />
      <h2>
        Microsoft
      </h2>
      <h4>
        100
      </h4>
      <h5
        className="colorBlue"
      >
        1
      </h5>
    </div>
    <div
      className="priceData"
    >
      <div
        className="row"
      >
        <div
          className="cell"
        >
          <div>
            Previous close
          </div>
          <div>
            99
          </div>
        </div>
        <div
          className="cell"
        >
          <div>
            Market Cap
          </div>
          <div>
            1000000
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="cell"
        >
          <div>
            52 Week Range
          </div>
          <div>
            0-100
          </div>
        </div>
        <div
          className="cell"
        >
          <div>
            Beta
          </div>
          <div>
            
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="cell"
        >
          <div>
            Dividend Yield
          </div>
          <div>
            NaN%
          </div>
        </div>
        <div
          className="cell"
        >
          <div>
            DCF
          </div>
          <div>
            
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="cell"
        >
          <div>
            Exchange
          </div>
          <div>
            
          </div>
        </div>
        <div
          className="cell"
        >
          <div>
            Country
          </div>
          <div>
            US
          </div>
        </div>
      </div>
    </div>
    <div
      className="info"
    >
      <h3>
        Company Information
      </h3>
      <div
        className="row"
      >
        <div
          className="cell"
        >
          <div>
            Sector:
          </div>
          <div>
            
          </div>
        </div>
        <div
          className="cell"
        >
          <div>
            Industry:
          </div>
          <div>
            
          </div>
        </div>
      </div>
      <h4>
        CEO
      </h4>
      <p />
      <h4>
        Website
      </h4>
      <p />
      <h4>
        Description
      </h4>
      <p />
    </div>
  </div>
</div>
`;
